src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -101,6 +101,20 @@ private IGeoMasterClient InitClient()
                     string value = CredentialTrapper.Obfuscate(item.Value);
                     appSettings.Add(item.Key, value);
                 }
+                else if(string.Compare(item.Key, "AzureWebJobsStorage", StringComparison.CurrentCulture) == 0)
+                {
+                    var storageAccountMatch = Regex.Match(item.Value.ToString(), @".*AccountName=(?<StorageAccount>.+?);.*");
```

Comment:
Instead of RegEx, do you want to use [CloudStorageAccount.Parse](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.cloudstorageaccount.parse?view=azure-dotnet) method instead?

Reply:
Also, are you trying to hide the storage account name only? AFAIK, account names themselves are not a concern. Are the keys for AzureWebJobsStorage already masked?

Reply:
No Puneet, I want to only show the storage account name and hide the key part in AzureWebJobsStorage. So for AzureWebJobsStorage in appsettings, we will return ***AccountName={StorageAccountName}**** instead of "******", which is what we are currently returning for AzureWebJobsStorage app settings. Currently there is no easy way to figure out which storage account the customer is using. Showing storage account name part in the value will be helpful in a lot of cases.

Reply:
I see. Just add `RegexOptions.IgnoreCase` to Regex.Match and the rest is fine

Reply:
@puneetg1983 do you know if the connection string value is case sensitive or not? I know the app settings key is case sensitive but not quite sure about the value. If it's case insensitive, add IgnoreCase option definitely makes sense:)